paper-submenu.html

```diff
@@ -0,0 +1,181 @@
+        </paper-menu>
+      </paper-submenu>
+      <paper-submenu disabled>
+        <paper-item class="menu-trigger">Unavailable</paper-item>
+        <paper-menu class="menu-content">
+          <paper-item>Disabled 1</paper-item>
+          <paper-item>Disabled 2</paper-item>
+        </paper-menu>
+      </paper-submenu>
+    </paper-menu>
+
+Just like in `<paper-menu>`, the focused item is highlighted, and the selected
+item has bolded text. Please see the `<paper-menu>` docs for which attributes
+(such as `multi` and `selected`), and styling options are available for the menu.
+
+@group Paper Elements
+@element paper-submenu
+@hero hero.svg
+@demo demo/index.html
+-->
+
+<dom-module id="paper-submenu">
+  <link rel="import" type="css" href="paper-menu-shared.css">
+
+  <template>
+    <div class="selectable-content" on-tap="_onTap">
+      <content id="trigger" select=".menu-trigger"></content>
+    </div>
+    <iron-collapse id="collapse" opened="{{opened}}">
+      <content select=".menu-content"></content>
+    </iron-collapse>
+  </template>
+
+</dom-module>
+
+<script>
+
+(function() {
+
+  Polymer({
+
+    is: 'paper-submenu',
+
+    properties: {
+      /**
+       * Fired when the submenu is opened.
+       *
+       * @event paper-submenu-open
+       */
+
+      /**
+       * Fired when the submenu is closed.
+       *
+       * @event paper-submenu-close
+       */
+
+      /**
+       * Set opened to true to show the collapse element and to false to hide it.
+       *
+       * @attribute opened
+       */
+      opened: {
+        type: Boolean,
+        value: false,
+        notify: true,
+        observer: '_openedChanged'
+      }
+    },
+
+    behaviors: [
+      Polymer.IronControlState
+    ],
+
+    get _parent() {
+      return Polymer.dom(this).parentNode;
+    },
+
+    get _trigger() {
+      return Polymer.dom(this.$.trigger).getDistributedNodes()[0];
+    },
+
+    attached: function() {
+      this.listen(this._parent, 'iron-activate', '_onParentIronActivate');
+    },
+
+    dettached: function() {
+      this.unlisten(this._parent, 'iron-activate', '_onParentIronActivate');
+    },
+
+    /**
+     * Make the dropdown content appear as an overlay positioned relative
+     * to the dropdown trigger.
+     */
+    open: function() {
+      if (this.disabled)
+        return;
+      this.$.collapse.show();
+    },
+
+    /**
+     * Hide the dropdown content.
+     */
+    close: function() {
+      this.$.collapse.hide();
+    },
+
+    _onTap: function() {
+      if (this.disabled)
+        return;
+      this.$.collapse.toggle();
+    },
+
+    _openedChanged: function(opened, oldOpened) {
+      if (opened) {
+        this.fire('paper-submenu-open');
+      } else if (oldOpened != null) {
+        this.fire('paper-submenu-close');
+      }
+    },
+
+    _onParentIronActivate: function(event) {
+      if (Polymer.Gestures.findOriginalTarget(event) != this._parent) {
```

Comment:
This is a little bike sheddy, but I consider the semantics of strict equality checking to be superior in most cases (exception: checking for `!= null`).